TestSuite/test_suite_base.py

```diff
@@ -0,0 +1,31 @@
+from abc import abstractmethod
+from pathlib import Path
+
+from demisto_sdk.commands.content_graph.interface import ContentGraphInterface
+from demisto_sdk.commands.content_graph.objects.base_content import BaseContent
+
+
+class TestSuiteBase:
+    def __init__(self, path: Path):
+        self.obj_path = path
+
+    @property
+    def object(self):
+        return BaseContent.from_path(self.obj_path)
+
+    def get_graph_object(self, interface: ContentGraphInterface):
+        return interface.from_path(self.obj_path)
+
+    @abstractmethod
+    def set_data(self, key_path_to_val: dict):
```

Comment:
why can't we implement it here?

Reply:
Because it dependent on the file structure YML or JSON so implemented in those classes

Reply:
I think a better solution is to have a property if this content item is JSON or YAML, and set data accordingly, instead implementing the same method in all classes

Reply:
as we discussed - due to this implemented only in the base classes (YML & JSON), the code will remain as is